src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -894,7 +894,7 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
 #endif
 
 #if defined(TARGET_LINUX) && !defined(HOST_ARM) && !defined(HOST_X86)
-    if (cacheSize == 0)
+    if (cacheSize == 0 || cacheSize == SIZE_MAX)
```

Comment:
since you are already filtering out `SIZE_MAX` in the `UPDATE_CACHE_SIZE_AND_LEVEL` macro, how can cacheSize be `SIZE_MAX`?